### PR TITLE
chore: move `isIOS` to `@vueuse/shared`

### PR DESCRIPTION
**Pull request by LittleSound**
#### Completed a [TODO](https://github.com/vueuse/vueuse/commit/930890a3860288806e1bcd67c77865df900741b8#diff-ee711f99aedfd3706e831df7c3f4578f8db0acbf17bc3990c888b28d6c4d2206R15):

```typescript
// TODO: move to @vueuse/share
const isIOS = /* #__PURE__ */ isClient && window?.navigator && window?.navigator?.platform && /iP(ad|hone|od)/.test(window?.navigator?.platform)

```
---

### What is the purpose of this pull request? 

- [ ] Bug fix
- [ ] New Feature
- [ ] Documentation update
- [x] Other

### Before submitting the PR, please make sure you do the following

- [x] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [x] Read the [Pull Request Guidelines](https://github.com/vueuse/vueuse/blob/main/packages/guidelines.md).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [ ] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [ ] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

**Comment by LittleSound** [comment]
Should we create a `usePlatform` function to help identify the platform?

File: packages/shared/utils/is.ts [hunk 1]
```diff
@@ -23,3 +23,4 @@ export const rand = (min: number, max: number) => {
   max = Math.floor(max)
   return Math.floor(Math.random() * (max - min + 1)) + min
 }
+export const isIOS = isClient && window?.navigator && window?.navigator?.userAgent && /iP(ad|hone|od)/.test(window?.navigator?.userAgent)
```
Comment (okxiaoliang4) [inline packages/shared/utils/is.ts:26]:
```suggestion
export const isIOS = /* #__PURE__ */ isClient && window?.navigator && window?.navigator?.userAgent && /iP(ad|hone|od)/.test(window?.navigator?.userAgent)
```

**Review by okxiaoliang4 — CHANGES_REQUESTED** [review]

File: packages/shared/utils/is.ts (hunk 1 above)
Comment (LittleSound) [reply packages/shared/utils/is.ts:26]: What does this PURE comment represent?

Comment (okxiaoliang4) [reply packages/shared/utils/is.ts]: > What does this PURE comment represent?

treeshake

File: packages/shared/utils/is.ts [hunk 2]
```diff
@@ -23,3 +23,4 @@ export const rand = (min: number, max: number) => {
   max = Math.floor(max)
   return Math.floor(Math.random() * (max - min + 1)) + min
 }
+export const isIOS = /* #__PURE__ */ isClient && window?.navigator?.userAgent && /iP(ad|hone|od)/.test(window.navigator.userAgent)
```
Comment (okxiaoliang4) [inline packages/shared/utils/is.ts:26]: Note. that the judgment condition has been changed from `navigator.platform` to `navigator.userAgent` 

`navigator.platform` is deprecated.

Comment (LittleSound) [reply packages/shared/utils/is.ts:26]: Sure, the new implementation has avoid to use `navigator.platform`

**Comment by wheatjs** [comment]
> Should we create a `usePlatform` function to help identify the platform?

Since it is not reactive we should just leave it as a utility like it is now

**Pull request merged**